app/daemon.go

```diff
@@ -130,15 +139,15 @@ func startManager(c *cli.Context) error {
 	}
 	kubeconfigPath := c.String(FlagKubeConfig)
 
-	if err := environmentCheck(); err != nil {
-		return errors.Wrap(err, "Failed environment check, please make sure you have iscsiadm/open-iscsi installed on the host")
-	}
-
 	currentNodeID, err := util.GetRequiredEnv(types.EnvNodeName)
 	if err != nil {
 		return fmt.Errorf("failed to detect the node name")
 	}
 
+	if err := environmentCheck(kubeconfigPath, currentNodeID); err != nil {
+		return errors.Wrap(err, "Failed environment check, please make sure you have iscsiadm/open-iscsi installed on the host")
```

Comment:
Since environmentCheck() checks everything, not just iscsi, this error message may be misleading.

Reply:
Agree with @james-munson's comment.
You can combine the errors in `environmentCheck` and just do
```
return errors.Wrap(err, "Failed environment check")
```
here.

Reply:
Modified. Thanks.